SeQuant/core/expr.hpp

```diff
@@ -369,25 +392,25 @@ class Expr : public std::enable_shared_from_this<Expr>,
   /// @return true if this object is of type @c T
   template <typename T>
   bool is() const {
-    if constexpr (std::is_same_v<std::decay_t<T>, Expr>)
+    if constexpr (is_expr_v<T>)
       return true;
     else
-      return this->type_id() == get_type_id<std::decay_t<T>>();
+      return this->type_id() == get_type_id<remove_cvref_t<T>>();
```

Comment:
There is a subtle difference between `remove_cvref ` and `decay` - see https://stackoverflow.com/a/74257596/3907364
But I think in this particular case, it doesn't matter (and since decay is not needed, I replaced it with the "less specialized" remove_cvref)